gobuffalo/azuredeploy.json

```diff
@@ -0,0 +1,196 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2014-04-01-preview/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+      "imageName" : {
+        "type": "String",
+        "defaultValue": "appsvc/sample-hello-world:latest",
+        "metadata": {
+          "description": "The image which defines the application to be hosted in Azure."
+        }
+      },
+      "name": {
+        "type": "String",
+        "defaultValue": "[concat('site', uniqueString(resourceGroup().id, deployment().name))]",
+        "metadata": {
+          "description": "A unique name to identify the site, and its relate assets once it is created."
+        }
+      },
+      "database": {
+        "type": "String",
+        "defaultValue": "none",
+        "allowedValues": [
+            "none",
+            "postgresql",
+            "mysql"
+        ],
+        "metadata": {
+          "description": "The flavor of database that should be deployed alongside this application."
+        }
+      },
+      "databaseName": {
+        "type": "String",
+        "defaultValue": "buffalo_development",
+        "metadata": {
+          "description": "The name that will identify the database which is created, should one be created."
+        }
+      },
+      "databaseAdministratorLogin": {
+          "type": "String",
+          "defaultValue": "[concat('admin', parameters('name'))]",
+          "metadata": {
+            "description": "The user handle for the administrator of the database to be created."
+          }
+      },
+      "databaseAdministratorLoginPassword": {
+        "type": "SecureString",
+        "defaultValue": "",
+        "metadata": {
+          "description": "The password for the administrator of the database to be created."
+        }
+      }
+    },
+    "variables": {
+      "hostingPlanName": "[concat('hostingPlan-', parameters('name'))]",
+      "postgresqlName": "[concat(parameters('name'), '-postgresql')]",
+      "mysqlName": "[concat(parameters('name'), '-mysql')]",
+      "postgresqlConnection": "[concat('postgres://', parameters('databaseAdministratorLogin'), '@', variables('postgresqlName'),':', parameters('databaseAdministratorLoginPassword'), '@', variables('postgresqlname'), '.postgres.database.azure.com:5432/', parameters('databaseName'))]",

```

Comment:
use reference() to get the uri

Reply:
For both instances of these comments, I believe that using `reference()` is insufficient for me, because I need to have the admin credentials embedded in the string.

Reply:
I'm not sure of that position though, let me know if there's something I'm missing.

Reply:
you still need to concat() the other things for the non-uri parts of the connect string... just can't hard-code the *database.azure.com piece.

Reply:
Looks like the `reference()` function can't be used in the variables section. https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-group-template-functions-resource#remarks-1

I could refactor my template so that the connection strings are computed inline, but given how much more verbose that will make the connection string line, I'd rather leave it alone.

Reply:
Sorry - not an option.